### PR TITLE
EQ FIR code re-organize

### PR DESCRIPTION
**Pull request by btian1**

### CODE REVIEW

File: src/audio/eq_fir/eq_fir_ipc4.c [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <kernel/abi.h>
+#include <user/eq.h>
+#include <user/fir.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
```
Comment (lyakh) [inline src/audio/eq_fir/eq_fir_ipc4.c:37]: something tells me that not all these headers are needed here. We do want to have all headers that are used in this file included explicitly, and it's difficult to keep the list perfect, but maybe you could do at least a light walk through and check if you find any obviously unused ones. E.g. uuid.h? platform.h? memory.h?

Comment (btian1) [reply src/audio/eq_fir/eq_fir_ipc4.c:37]: that's correct, just need takes time

File: zephyr/CMakeLists.txt [hunk 2]
```diff
@@ -427,12 +427,34 @@ zephyr_library_sources_ifdef(CONFIG_COMP_FIR
...
 elseif(CONFIG_IPC_MAJOR_4)
+	zephyr_library_sources_ifdef(CONFIG_COMP_FIR
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_hifi3.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_hifi2ep.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_generic.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_ipc4.c
+		${SOF_MATH_PATH}/fir_generic.c
+		${SOF_MATH_PATH}/fir_hifi2ep.c
+		${SOF_MATH_PATH}/fir_hifi3.c
```
Comment (marc-hb) [inline zephyr/CMakeLists.txt:455]: This is the same list twice! Why copy/paste/diverge?

**Review by marc-hb — CHANGES_REQUESTED** [review]
This PR adds eight new C files to `zephyr/CMakeLists.txt` but it does not seem to remove them from anywhere else. So how is this a "re-org"? Please explain in the commit message what is happening here.

File: zephyr/CMakeLists.txt (hunk 2 above)
Comment (btian1) [reply zephyr/CMakeLists.txt:455]: for separate ipc3/ipc4 code, if you have other good ways, I can have a try.

Comment (marc-hb) [reply zephyr/CMakeLists.txt:455]: After a second look I see only one file difference. So only that file should be under CONFIG_IPC_. No copy/paste required for the others.

File: src/audio/eq_fir/eq_fir_ipc3.c [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2017 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+//         Liam Girdwood <liam.r.girdwood@linux.intel.com>
+//         Keyon Jie <yang.jie@linux.intel.com>
+
+#include "eq_fir.h"
```
Comment (lyakh) [inline src/audio/eq_fir/eq_fir_ipc3.c:9]: no, you have to include all the headers, that this file uses here explicitly. We don't use gather-all headers

File: src/audio/eq_fir/eq_fir_ipc4.c [hunk 4]
```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2017 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+//         Liam Girdwood <liam.r.girdwood@linux.intel.com>
+//         Keyon Jie <yang.jie@linux.intel.com>
+
+#include "eq_fir.h"
```
Comment (lyakh) [inline src/audio/eq_fir/eq_fir_ipc4.c:9]: ditto

**Review by lyakh — CHANGES_REQUESTED** [review]

File: src/audio/eq_iir/CMakeLists.txt [hunk 5]
```diff
@@ -5,4 +5,4 @@ if(CONFIG_IPC_MAJOR_3)
 	add_local_sources(sof eq_iir_ipc3.c)
 elseif(CONFIG_IPC_MAJOR_4)
 	add_local_sources(sof eq_iir_ipc4.c)
-endif()
+endif()
```
Comment (marc-hb) [inline src/audio/eq_iir/CMakeLists.txt:8]: Is your editor misconfigured?
- It removed the EOL
- It didn't show that to you in git status

File: src/audio/eq_fir/eq_fir_ipc3.c (hunk 3 above)
Comment (lyakh) [reply src/audio/eq_fir/eq_fir_ipc3.c:13]: I think you're still missing some headers (see below)

File: src/audio/eq_fir/eq_fir_ipc3.c [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+//         Liam Girdwood <liam.r.girdwood@linux.intel.com>
+//         Keyon Jie <yang.jie@linux.intel.com>
+
+#include <sof/audio/module_adapter/module/generic.h>
+#include <ipc/stream.h>
+#include <sof/audio/component.h>
+
+#include "eq_fir.h"
+
+LOG_MODULE_DECLARE(eq_fir, CONFIG_SOF_LOG_LEVEL);
```
Comment (lyakh) [inline src/audio/eq_fir/eq_fir_ipc3.c:15]: this needs a header

File: src/audio/eq_fir/eq_fir_ipc3.c [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+#endif /* CONFIG_FORMAT_S24LE */
+#if CONFIG_FORMAT_S32LE
+	case SOF_IPC_FRAME_S32_LE:
+		comp_dbg(mod->dev, "set_fir_func(), SOF_IPC_FRAME_S32_LE");
+		set_s32_fir(cd);
+		break;
+#endif /* CONFIG_FORMAT_S32LE */
+	default:
+		comp_err(mod->dev, "set_fir_func(), invalid frame_fmt");
+		return -EINVAL;
```
Comment (lyakh) [inline src/audio/eq_fir/eq_fir_ipc3.c:42]: errno.h

**Review by lyakh — APPROVED** [review]

File: src/audio/eq_fir/eq_fir_ipc4.c [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+#include <sof/audio/audio_stream.h>
+#include <sof/list.h>
+#include <sof/trace/trace.h>
+#include <errno.h>
+
+#include "eq_fir.h"
+
+LOG_MODULE_DECLARE(eq_fir, CONFIG_SOF_LOG_LEVEL);
+
+int set_fir_func(struct processing_module *mod, enum sof_ipc_frame fmt)
```
Comment (singalsu) [inline src/audio/eq_fir/eq_fir_ipc4.c:22]: I haven't earlier needed different processing function select for ipc versions. Are the source and sink params set from init() IPC before selecting the processing function?

File: src/audio/eq_fir/eq_fir_ipc4.c [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+	struct sof_ipc_stream_params *params = mod->stream_params;
+	struct sof_ipc_stream_params comp_params;
+	struct comp_dev *dev = mod->dev;
+	struct comp_buffer *sinkb;
+	enum sof_ipc_frame valid_fmt, frame_fmt;
+	int i, ret;
+
+	comp_dbg(dev, "eq_fir_params()");
+
+	comp_params = *params;
```
Comment (singalsu) [inline src/audio/eq_fir/eq_fir_ipc4.c:62]: There's a nicer source and sink formats set helper available. E.g.

```
static void eq_fir_params(struct processing_module *mod)
{
	struct sof_ipc_stream_params *params = mod->stream_params;
	struct comp_buffer *sinkb, *sourceb;
	struct comp_dev *dev = mod->dev;

	ipc4_base_module_cfg_to_stream_params(&mod->priv.cfg.base_cfg, params);
	component_set_nearest_period_frames(dev, params->rate);

	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
	ipc4_update_buffer_format(sourceb, &mod->priv.cfg.base_cfg.audio_fmt);

	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
	ipc4_update_buffer_format(sinkb, &mod->priv.cfg.base_cfg.audio_fmt);
}
```

**Review by lgirdwood — APPROVED** [review]
LGTM, lets address @singalsu and I think we are good to merge.

**Review by singalsu — APPROVED** [review]
Looks good to me, thanks for the clean up work!

**Pull request merged**